### ISSUE TITLE
Exception saving the result of 'unionAll()'  breaks the storage and requires to restart.  (java.lang.ClassCastException)

### ISSUE THREAD

**Issue by NicolasJuanF** [orientechnologies/orientdb#9781]
## Expected behavior   
To get a validation exception and keep using the database normally  

## Actual behavior  
update throws an exception and requires to restart.
```
Exception `646F00A5` in storage `plocal:/opt/orientdb-community-3.2.5/databases/demo`: 3.2.5 (build c4298657c01683192ba0b7bfffdf82226c164506, branch UNKNOWN)
java.lang.ClassCastException: class com.orientechnologies.common.collection.OMultiCollectionIterator cannot be cast to class java.util.Collection (com.orientechnologies.common.collection.OMultiCollectionIterator is in unnamed module of loader 'app'; java.util.Collection is in module java.base of loader 'bootstrap')
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValue(ORecordSerializerBinaryV1.java:991)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.writeEmbeddedMap(ORecordSerializerBinaryV1.java:672)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValue(ORecordSerializerBinaryV1.java:1017)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValues(ORecordSerializerBinaryV1.java:404)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeDocument(ORecordSerializerBinaryV1.java:462)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serialize(ORecordSerializerBinaryV1.java:483)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.toStream(ORecordSerializerBinary.java:132)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commitEntry(OAbstractPaginatedStorage.java:5946)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:2367)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OEnterpriseLocalPaginatedStorage.commit(OEnterpriseLocalPaginatedStorage.java:1133)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:2190)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalCommit(ODatabaseDocumentEmbedded.java:1978)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:621)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:114)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.commit(ODatabaseDocumentAbstract.java:1593)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.commit(ODatabaseDocumentAbstract.java:1563)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.saveAll(ODatabaseDocumentEmbedded.java:1272)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:208)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.saveInternal(ODatabaseDocumentAbstract.java:1484)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:1432)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:1315)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:78)
	at com.orientechnologies.orient.core.sql.executor.SaveElementStep$1.next(SaveElementStep.java:37)
	at com.orientechnologies.orient.core.sql.executor.CountStep.syncPull(CountStep.java:52)
	at com.orientechnologies.orient.core.sql.executor.OSelectExecutionPlan.fetchNext(OSelectExecutionPlan.java:34)
	at com.orientechnologies.orient.core.sql.executor.OUpdateExecutionPlan.executeInternal(OUpdateExecutionPlan.java:41)
	at com.orientechnologies.orient.core.sql.executor.ScriptLineStep.syncPull(ScriptLineStep.java:33)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.doExecute(OScriptExecutionPlan.java:78)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.fetchNext(OScriptExecutionPlan.java:39)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.fetchNext(OLocalResultSet.java:36)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.<init>(OLocalResultSet.java:27)
	at com.orientechnologies.orient.core.command.OSqlScriptExecutor.executeInternal(OSqlScriptExecutor.java:126)
	at com.orientechnologies.orient.core.command.OSqlScriptExecutor.execute(OSqlScriptExecutor.java:50)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.execute(ODatabaseDocumentEmbedded.java:722)
	at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(OServerCommandPostBatch.java:211)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:242)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:800)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:68)
$ANSI{green {db=demo}} Error on transaction commit `646F00A5`
java.lang.ClassCastException: class com.orientechnologies.common.collection.OMultiCollectionIterator cannot be cast to class java.util.Collection (com.orientechnologies.common.collection.OMultiCollectionIterator is in unnamed module of loader 'app'; java.util.Collection is in module java.base of loader 'bootstrap')
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValue(ORecordSerializerBinaryV1.java:991)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.writeEmbeddedMap(ORecordSerializerBinaryV1.java:672)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValue(ORecordSerializerBinaryV1.java:1017)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeValues(ORecordSerializerBinaryV1.java:404)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serializeDocument(ORecordSerializerBinaryV1.java:462)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV1.serialize(ORecordSerializerBinaryV1.java:483)
	at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.toStream(ORecordSerializerBinary.java:132)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commitEntry(OAbstractPaginatedStorage.java:5946)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:2367)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.OEnterpriseLocalPaginatedStorage.commit(OEnterpriseLocalPaginatedStorage.java:1133)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:2190)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalCommit(ODatabaseDocumentEmbedded.java:1978)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:621)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:114)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.commit(ODatabaseDocumentAbstract.java:1593)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.commit(ODatabaseDocumentAbstract.java:1563)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.saveAll(ODatabaseDocumentEmbedded.java:1272)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:208)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.saveInternal(ODatabaseDocumentAbstract.java:1484)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:1432)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:1315)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:78)
	at com.orientechnologies.orient.core.sql.executor.SaveElementStep$1.next(SaveElementStep.java:37)
	at com.orientechnologies.orient.core.sql.executor.CountStep.syncPull(CountStep.java:52)
	at com.orientechnologies.orient.core.sql.executor.OSelectExecutionPlan.fetchNext(OSelectExecutionPlan.java:34)
	at com.orientechnologies.orient.core.sql.executor.OUpdateExecutionPlan.executeInternal(OUpdateExecutionPlan.java:41)
	at com.orientechnologies.orient.core.sql.executor.ScriptLineStep.syncPull(ScriptLineStep.java:33)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.doExecute(OScriptExecutionPlan.java:78)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.fetchNext(OScriptExecutionPlan.java:39)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.fetchNext(OLocalResultSet.java:36)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.<init>(OLocalResultSet.java:27)
	at com.orientechnologies.orient.core.command.OSqlScriptExecutor.executeInternal(OSqlScriptExecutor.java:126)
	at com.orientechnologies.orient.core.command.OSqlScriptExecutor.execute(OSqlScriptExecutor.java:50)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.execute(ODatabaseDocumentEmbedded.java:722)
	at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(OServerCommandPostBatch.java:211)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:242)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:800)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:68)
```

## Steps to reproduce  

```sql
create class example extends V;
create property example.metadata EMBEDDEDMAP;

let $example = create vertex example;
let $a = {"aKey":"aValue"};
let $b = {"anotherKey":"anotherValue"};
let $u = unionAll($a, $b); 

/* both of the following throw the exception and require to restart the server*/
update $example set metadata["something"] = $u;
update $example set metadata.something = $u;
```

**Comment by tglman** [comment]
Hi, 

Added a fix for the case, will be released with 3.2.8.

Regards

**Comment by tglman** [comment]
Hi,

A fix for this case has been rle
